src/modules/common/components/ScrollingSearchBox.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import style from './ScrollingSearchBox.css'
+import { animateScroll } from 'react-scroll'
+import SearchInput from './SearchInput'
+
+export class ScrollingSearchBox extends React.Component {
+  static propTypes = {
+    filterText: PropTypes.string.isRequired,
+    onFilterTextChange: PropTypes.func.isRequired,
+    stickyTop: PropTypes.number.isRequired,
+    spaceSearch: PropTypes.bool,
+    children: PropTypes.element.isRequired
+  }
+
+  static defaultProps = {
+    stickyTop: 0
+  }
+
+  onClick () {
+    const documentScroll = document.documentElement.scrollTop
+    const elementTop = this._node.offsetTop
+    if (documentScroll < elementTop) {
+      this.scroll()
+    }
+  }
+
+  scroll () {
+    const elementTop = this._node.offsetTop
+    animateScroll.scrollTo(elementTop, {duration: 500})
+  }
+
+  onFilterTextChange (value) {
+    this.props.onFilterTextChange(value)
+    this.scroll()
+  }
+
+  setReference (node) {
+    if (node) {
+      this._node = node
+    }
+  }
+
+  render () {
+    return <React.Fragment>
+      <div ref={node => this.setReference(node)} className={style.searchBar}
```

Comment:
Refs are not that good to use in react. I don't see an other option here. The whole scrolling stuff is also a little bit overpowered maybe here :)

Reply:
Yeah, but we need ref here.